libbeat/publisher/processing/processors.go

```diff
@@ -180,7 +180,7 @@ func debugPrintProcessor(info beat.Info, log *logp.Logger) *processorFn {
 			return event, nil
 		}
 
-		log.Debugf("Publish event: %s", b)
+		logp.Debug("publish", "Publish event: %s", b)
```

Comment:
Oh I see. The reason we don't use `logp.Debug` is that the publisher pipeline is used in 2 places. For outputs, but also for monitoring. By introducing `log.Debugf` we added structured logging and per package selector capabilities to the pipeline. That is we currently have 2 selectors `publisher`, and `monitoring`. Each with it's own pipeline debug selectors.

If possible we should not use the global loggers defined in logp.

The `publisher` selector comes from this place: https://github.com/elastic/beats/blob/master/libbeat/cmd/instance/beat.go#L321

The `monitoring` selector comes from this place: https://github.com/elastic/beats/blob/master/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L116